File: Server/microservices/user-service/src/main/java/com/taskman/user_service/service/interfaces/UserService.java
package com.taskman.user_service.service.interfaces;

import com.taskman.user_service.dto.TeamDTO;
import com.taskman.user_service.dto.UserDTO;
import com.taskman.user_service.dto.request.CreateUserRequest;
import com.taskman.user_service.dto.request.UpdateUserRequest;
import com.taskman.user_service.entity.enums.UserRole;

import java.util.List;

public interface UserService {
    // Create
    UserDTO createUser(CreateUserRequest request);

    // Read
    UserDTO getUserById(Long id);
    UserDTO getUserByEmail(String email);
    List<UserDTO> getAllUsers();
    boolean existsByEmail(String email);

    // Update
    UserDTO updateUser(Long id, UpdateUserRequest request);
    UserDTO updateUserRole(Long id, UserRole role);
    UserDTO activateUser(Long id);
    UserDTO deactivateUser(Long id);

    // Delete
    void deleteUser(Long id);

    // Team related
    List<TeamDTO> getUserTeams(Long userId);
}

File: taskman_front/src/services/UserService.js
import api from "../utils/axios";

export const UserService = {
  async getAllUsers() {
    const response = await api.get("/users");
    return response.data;
  },

  async createUser(userData) {
    const response = await api.post("/users", userData);
    return response.data;
  },

  async updateUser(userId, userData) {
    const response = await api.put(`/users/${userId}`, userData);
    return response.data;
  },

  async deleteUser(userId) {
    const response = await api.delete(`/users/${userId}`);
    return response.data;
  },

  async getUser(userId) {
    const response = await api.get(`/users/${userId}`);
    return response.data;
  },
};

export default UserService;
